api/errors/errors.go

```diff
@@ -1,47 +1,57 @@
 package errors
 
-import "net/http"
+// NotFoundError signals that the requested object doesn't exist
+type NotFoundError interface {
+	error
+	NotFound()
+}
 
-// apiError is an error wrapper that also
-// holds information about response status codes.
-type apiError struct {
+// InvalidParameterError signals that the user input is invalid
+type InvalidParameterError interface {
```

Comment:
@cpuguy83 Yes, with some advice like "don't use these directly unless those functions don't work for you".